Incentive module multiple reward denom tests

denalimarsh opened the pull request:
Updates the Incentive module's test suite for multiple reward denoms.

karzak commented on x/incentive/keeper/payout_test.go:
```diff
@@ -153,179 +154,189 @@ func (suite *KeeperTestSuite) TestPayoutUSDXMintingClaim() {
+				contains:   "",
+			},
+		},
+		{
```
Add test with multiple coins on the claim, add similar test for USDX minting rewards

denalimarsh replied:
Updated payout test in https://github.com/Kava-Labs/kava/pull/788/commits/c63ff89829ca9b789374f3999e0a928104020ecd.

What do you mean by 'similar test for USDX minting rewards'? USDXMintingClaims don't use `MultiRewardPeriod`/`MultiRewardIndex` yet. If the suggestion here is to test multiple reward coins within the current set up (e.g. USDXMinting RewardPeriods params contain two RewardPeriods for bnb/xrp that have different reward denoms such as ukava/hard) I'd rather just update USDXMintingClaims to use `MultiRewardPeriods` and then test multiple reward denoms using the existing testing process from TestPayoutHardLiquidityProviderClaim. I can make an issue for it.

karzak replied:
Ah my bad, forgot we didn't implement Multi rewards for USDX minting yet. Ignore.

karzak commented on x/incentive/keeper/rewards.go:
```diff
@@ -521,6 +523,7 @@ func (k Keeper) SynchronizeHardDelegatorRewards(ctx sdk.Context, delegator sdk.A
 	}
 
 	// Add rewards to delegator's hard claim
+	// TODO: types.HardLiquidityRewardDenom
```
merge artifact I think- can delete

denalimarsh replied:
Removed in https://github.com/Kava-Labs/kava/pull/788/commits/8d1cc4a9f8a912b2cfec44b1803c3aaa93c6ff57

karzak reviewed (changes requested):
Looks really good.

Add multiple coins payout tests, review merge changes, otherwise LGTM. Feel free to merge once new tests are passing and added.

denalimarsh commented:
A bunch of tests and some functionality got deleted by the master merge so I added them back and updated the master tests in https://github.com/Kava-Labs/kava/pull/788/commits/bfaa1b2f2a63bf80c01432d0e93687345586e616. Hopefully nothing else got accidentally removed, currently the tests are passing and everything seems to be working.
